### ISSUE TITLE
CMake package has hardcoded path to ZeroMQ library

### ISSUE THREAD

**Issue by sdebionne** [slsdetectorgroup/slsDetectorPackage#234]
##### *Distribution: 

Conda, but probably any that provides CMake package.

##### *Detector type: 

Any

##### *Software Package Version: 

v5.0.1

##### Priority:

Low

##### *Describe the bug

When installing the CMake package (e.g. from Conda), the targets definition contains an hardcoded path (set at build time) to the ZeroMQ library.

Here is what I get in `slsdetectorpackage-targets.cmake` after building and installing the Conda package locally:

``` 
set_target_properties(slsSupportObject PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "slsProjectOptions;/nobackup/lid29pwr9/debionne/miniconda3/conda-bld/sls_detector_software_1607093377355/_build_env/lib/libzmq.so;rapidjson;\$<LINK_ONLY:slsProjectWarnings>>)
```

Later when using the package with `find_package(slsDetectorPackage REQUIRED)`, I get an obvious link error like

```
ninja: error: '/nobackup/lid29pwr9/debionne/miniconda3/conda-bld/sls_detector_software_1607093377355/_build_env/lib/libzmq.so', needed by 'detectors/psi/src/libpsi_ctrl.so', missing and no known rule to make it
```

##### Expected behavior

Use `find_dependency` (like for HDF5) in the `project-config.cmake.in`. Since `FindZeroMq.cmake` is not builtin, it probably needs to be installed as well.

**Comment by sdebionne** [comment]
I should have mentioned that a trivial fix is to edit the installed `slsdetectorpackage-targets.cmake` and remove the path:

```
set_target_properties(slsSupportObject PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "slsProjectOptions;zmq;rapidjson;\$<LINK_ONLY:slsProjectWarnings>>)
```

**Comment by erikfrojdh** [comment]
Ok I see the problem. We are using a FindZeroMQ adpopted from azmq which is responsible for writing the hard coded path, when use with ${ZeroMQ_LIBRARIES}. 

Since newer versions of libzmq includes support for find_package(ZeroMQ) I will try the same strategy as cppzmq, first trying that and if not using our own...

**Comment by erikfrojdh** [comment]
@sdebionne Could you have a look at the linked PR since I think you have more experience with CMake. If this looks reasonable I'll merge and include this fix for 5.1.0
